PokeDemoTests/PokemonListPresenterTests.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  PokeDemoTests.swift
+//  PokeDemoTests
+//
+//  Created by Rudr Bansal on 06/08/22.
+//
+
+import XCTest
+@testable import PokeDemo
+import Alamofire
+
+final class PokemonListPresenterTests: XCTestCase {
+    
+    var presenter: PokemonListPresenter!
+    
+    // MARK: - Lifecycle
+    
+    override func setUp() {
+        super.setUp()
+        presenter = PokemonListPresenter(service: Service.shared)
+    }
+    
+    override func tearDown() {
+        presenter = nil
+        super.tearDown()
+    }
+    
+    // MARK: - Tests
+    
+    func testDelegateSetsValidValue(){
+        // Given
+        let mockDelegate = MockPresenterDelegate()
+        
+        // When
+        presenter.setViewDelegate(delegate: mockDelegate)
+        
+        // Then
+        XCTAssertTrue(presenter.delegate is MockPresenterDelegate)
+    }
+    
+    func testGetPokemonListCallsAPI(){
+        // Given
+        let mockService = MockService()
+        
+        // When
+        presenter = PokemonListPresenter(service: mockService)
```

Comment:
If you inject MockService in setup. You don't need to do it in every tests. 

Read XCTest lifecycle - 

http://trikalabs.com/xctestcase-lifecycle/

Reply:
Or better as mentioned in the block, you can setup presenter in class setup and deinit in class teardown

Reply:
Covered in 95b810a9d180840d001ce309cd07301a9c2284de